src/main/scala/io/delta/DeltaMergeBuilder.scala

```diff
@@ -80,36 +252,111 @@ case class DeltaMergeBuilder private[delta](
   }
 }
 
-case class DeltaMergeMatchedActionBuilder private[delta](
+object DeltaMergeBuilder {
+  /**
+   * :: Unstable ::
+   *
+   * Private method for internal usage only. Do not call this directly.
+   */
+  @Unstable
+  private[delta] def apply(
+      targetTable: DeltaTable,
+      source: DataFrame,
+      onCondition: Column): DeltaMergeBuilder = {
+    new DeltaMergeBuilder(targetTable, source, onCondition, Nil)
+  }
+}
+
+/**
+ * :: Evolving ::
+ *
+ * Builder class to specify the actions to perform when a target table row has matched a
+ * source row based on the given merge condition and optional match condition.
+ *
+ * See [[DeltaMergeBuilder]] for more information.
+ *
+ * @since 0.3.0
+ */
+@Evolving
+class DeltaMergeMatchedActionBuilder private(
     private val mergeBuilder: DeltaMergeBuilder,
     private val matchCondition: Option[Column]) {
 
+  /**
+   * :: Evolving ::
+   *
+   * Update the matched table row based on the rules defined by `set`.
+   *
+   * @param set rules to update a row as a Scala map between target column names and
+   *            corresponding expressions as Column objects.
```

Comment:
rules to update a row as a Scala map between target column names and
their corresponding update expressions as Column objects.

Reply:
The above wording suggestions apply to the overloading APIs below.